### PR TITLE
Add method metrics reporter

### PR DESCRIPTION
**Pull request by SomeHats**

### CODE REVIEW

**Comment by dotbear** [comment]
Approved !

[Approved with PullApprove](https://pullapprove.com/musicglue/mg-express/pull-request/22/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=musicglue/mg-express)

File: src/methodMetrics.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        .then(() => 'resolved')
+        .catch(() => 'rejected')
+        .then(status => {
+          const duration = Date.now() - start;
+          const tags = [
+            `status:${status}`,
+            `method:${instanceName}/${methodName}`,
+          ];
+          metrics.count(`node.${key}.call`, 1, tags);
+          metrics.guage(`node.${key}.time`, duration, tags);
```
Comment (nikcorg) [inline src/methodMetrics.js]: Should that be `gauge`?

File: src/methodMetrics.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import * as metrics from './metrics';
+
+export default (instance, key, instanceName, methods) => {
+  methods.forEach(methodName => {
+    const originalMethod = instance[methodName];
+    instance[methodName] = function (...args) {
+      const start = Date.now();
+      const result = originalMethod.apply(this, args);
+
+      Promise.resolve(result)
```
Comment (nikcorg) [inline src/methodMetrics.js:10]: Missing return?

Comment (nikcorg) [inline src/methodMetrics.js:10]: Nevermind. Misread the function.

File: src/methodMetrics.js (hunk 1 above)
Comment (SomeHats) [inline src/methodMetrics.js]: holy shit i can't spell at all

**Comment by nikcorg** [comment]
Approved pending spelling fixes.

[Approved with PullApprove](https://pullapprove.com/musicglue/mg-express/pull-request/22/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=musicglue/mg-express)

**Pull request merged**